### PR TITLE
Pull Request for branch react-capstone that contains full functionality for Veejayz Gen-Store

### PR DESCRIPTION
**Pull request by jkanyi-web**
- Added redux store that connects redux to our app
- Added itemsSlice.js that defines a Redux slice named 'items' that manages the state for loading items from an external API using asynchronous actions, with loading, items, and error properties in the state.
- Created a file items.js that defines a React component named 'Items' that displays a list of items fetched from an external API, allows users to filter items by a search term, and provides links to view individual item details. The component relies on Redux state management and React Router for navigation.
- Created a itemDetail.js file that defines a React component named 'ItemDetail' that displays detailed information about a selected item. It retrieves the item ID from the URL using React Router, uses Redux to fetch items from an external API, and then displays the details of the selected item.
- Created a Header.js file that defines a simple React functional component named 'Header' that renders a navigation bar with a single link pointing to the root URL and displaying the text 'Veejayz-Hub'.
- Created tests for various components in the project

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @jkanyi-web  🙋‍♀️,  

While you made a great effort in this project, I can't review your code.

## Invalid Code Review Request
 
You have submitted a project that has a README that is not professional. Please document the README using the new microverse [README template](https://github.com/microverseinc/readme-template) and make sure that all sections are customized to the projects relevant content.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by rloterh** [comment]
Hi @jkanyi-web 

Kindly ensure to remove/update all similar to the underlined from your ReadMe asap!

image

File: src/index.css [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+img {
+  width: 150px;
+  height: 150px;
+}
+
+@media all and (max-width: 768px) {
+  .overall-container {
+    grid-template-columns: repeat(1, minmax(0, 1fr));
+  }
```
Comment (rloterh) [inline src/index.css]: - [ ] Good job so far, Kindly note, however, we as much as possible want your design to match the provided [template](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). Let's therefore ensure that the cards are displaying 2 on a row on mobile screens and follow the alternating color pattern on the homepage similar to the [original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). Difference between background color of each card and the cards opposite on horizontal and vertical sides should be lighter and darker theme of the chosen color, like seen in the original design.


**Your design:**        |  **The origin design:**
:-------------------------:|:-------------------------:
Your desgin |  Home Page

File: src/index.css [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+.item-details {
+  display: flex;
+  flex-direction: row;
+  gap: 5%;
+}
+
+.item-details-div {
+  margin-top: 10%;
```
Comment (rloterh) [inline src/index.css]: - [ ] Similarly, let's ensure that the details page matches the [original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). The layout/positioning of elements and color pattern closely matches the [template design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)).

**Your design:**        |  **The origin design:**
:-------------------------:|:-------------------------:
Your desgin |  Home Page

File: src/__tests__/items.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable import/no-named-as-default */
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import store from '../redux/store';
+import '@testing-library/jest-dom';
+import Items from '../redux/features/Items';
```
Comment (rloterh) [inline src/__tests__/items.test.js:7]: - [ ] Test cases are failing! Let's ensure that the test dependencies are properly configured, and all test cases are passing.
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Kanyi, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there!  
🏁  🌴 🚦 🚗 🏫  🏡   _🏃 

## Highlights 👍  
- Nice job on presenting your project!
- SPA feature working as intended!
- Properly implemented state management using Redux!
- Filter properly implemented!
- Good job with the tests!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @jkanyi-web,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Well-organized and clean project ✔️
- Correct workflow was used ✔️
- Your deployed app works fine ✔️
- Professional readme file ✔️
- No linter errors ✔️

apprrrr

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**